src/components/templates/PartyMap/components/Map/hooks/usePartygoersOverlay.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useMemo } from "react";
+
+import { User } from "types/User";
+import { ReactHook } from "types/utility";
+
+import { WithId } from "utils/id";
+import { isTruthy } from "utils/types";
+
+import { MapPartygoerOverlay } from "components/molecules/MapPartygoerOverlay";
+
+interface UsePartygoersOverlay {
+  showGrid?: boolean;
+  userUid?: string;
+  venueId: string;
+  withMiniAvatars?: boolean;
+  rows: number;
+  columns: number;
+  partygoers?: readonly WithId<User>[];
+  setSelectedUserProfile: (user: WithId<User>) => void;
+}
+
+export type UsePartygoersReturn =
+  | ("" | JSX.Element)[]
+  | JSX.Element
+  | undefined;
+
+export const usePartygoersOverlay: ReactHook<
+  UsePartygoersOverlay,
+  UsePartygoersReturn
+> = ({
+  showGrid,
+  userUid,
+  venueId,
+  withMiniAvatars,
+  rows,
+  columns,
+  partygoers,
+  setSelectedUserProfile,
+}) => {
+  return useMemo(() => {
+    // @debt partygoers can be undefined because our types are broken so check explicitly
+    if (!showGrid || !partygoers || !userUid) return <div />;
+
+    // @debt workaround, sometimes partygoers are duplicated but the new ones don't have id's
+    const filteredPartygoers = partygoers.filter(
+      (partygoer) => partygoer && isTruthy(partygoer.id)
+    );
```

Comment:
> - `@debt` partygoers can be undefined because our types are broken so check explicitly
> - `@debt` workaround, sometimes partygoers are duplicated but the new ones don't have id's